src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/PrefetchGetRecordsCache.java

```diff
@@ -69,47 +72,55 @@ public GetRecordsResult getNextResult() {
 
     @Override
     public void shutdown() {
-        executorService.shutdown();
+        executorService.shutdownNow();
     }
-    
+
     private class DefaultGetRecordsCacheDaemon implements Runnable {
         @Override
         public void run() {
             while (true) {
+                if (Thread.interrupted()) {
+                    log.warn("Prefetch thread was interrupted.");
+                    break;
+                }
                 if (prefetchCounters.byteSize < maxByteSize && prefetchCounters.size < maxRecordsCount) {
```

Comment:
It may make sense to provide the counters with the limits, and just have a method like `shouldRetrieveData()`